Tests/SpeechToTextV1Tests/SpeechToTextTests.swift

```diff
@@ -73,6 +89,196 @@ class SpeechToTextTests: XCTestCase {
+        let failToCreateCustomization =  { (error: Error) in
+            XCTFail("Failed to create the new customization: \(error)")
+            return
+        }
+        
+        let description = "Create a new customization."
+        let expectation = self.expectation(description: description)
+        speechToText.createCustomization(
+            withName: self.trainedCustomizationName,
+            withBaseModelName: "en-US_BroadbandModel",
+            description: self.trainedCustomizationDescription,
+            failure: failToCreateCustomization) { customization in
+                
+            self.trainedCustomizationID = customization.customizationID
+            expectation.fulfill()
+        }
+        waitForExpectations()
+    }
+    
+    func addCorpusToTrainedCustomization() {
+        
+        let failToCreateCorpus =  { (error: Error) in
+            XCTFail("Failed to create the new corpus: \(error)")
+            return
+        }
+        
+        guard let corpusFile = loadFile(name: "healthcare", withExtension: "txt") else {
+            XCTFail("Failed to load file needed to create the corpus.")
+            return
+        }
+        
+        let description = "Add a corpus for the customization."
+        let expectation = self.expectation(description: description)
+        speechToText.addCorpus(
+            withName: corpusName,
+            fromFile: corpusFile,
+            customizationID: self.trainedCustomizationID,
+            failure: failToCreateCorpus) {
+                
+            expectation.fulfill()
+        }
+        waitForExpectations()
+    }
+    
+    func trainCustomizationWithCorpus() {
+        
+        let failToGetCorpusStatus = { (error: Error) in
+            XCTFail("Failed to get the status of the new corpus: \(error)")
+            return
+        }
+        
+        let failToTrainCustomization = { (error: Error) in
+            XCTFail("Failed to train the new customization: \(error)")
+            return
+        }
+        
+        let failToGetCustomizationStatus = { (error: Error) in
+            XCTFail("Failed to get the status of the new customization: \(error)")
+            return
+        }
+        
+        var trained = false
+        var tries = 0
+        while(!trained) {
+            tries += 1
+            let description = "Wait until the corpus is processed before training the customization."
+            let expectation = self.expectation(description: description)
+            
+            speechToText.getCorpus(
+                withName: corpusName,
+                customizationID: self.trainedCustomizationID,
+                failure: failToGetCorpusStatus) { corpus in
+                    
+                    if corpus.status == .analyzed {
+                        trained = true
+                    }
+                    expectation.fulfill()
+            }
+            waitForExpectations()
+            
+            if tries > 10 {
```

Comment:
That's true. Adding the check to fail immediately if ```corpus.status == .undetermined``` is a good call. 

Yeah I can't think of a way around the `if tries > 10` check if our test cases don't keep track of how much time has passed per test. 

would it instead be something like this?
```swift
repeat {
     speechToText.getCorpus(
        ...
     ) { corpus in 
        switch corpus.status {
        case .analyzed: 
             rained = true
             expectation.fulfill()
        case .undetermined: XCTFail("Corpus trained unsuccessfully"); break
        ...
        }
        expectation.fulfill()
    }
} while (tries > 10 && corpus.status != .processed)
```